src/olympia/blocklist/tasks.py

```diff
@@ -143,16 +148,37 @@ def import_block_from_blocklist(record):
 
 
 @task
-def upload_filter_to_kinto(generation_time):
+def upload_filter_to_kinto(generation_time, is_base=True, upload_stash=False):
     server = KintoServer(
         KINTO_BUCKET, KINTO_COLLECTION_MLBF, kinto_sign_off_needed=False)
+    mlbf = MLBF(generation_time)
+    # Deal with possible stashes first
+    if upload_stash:
+        # If we have a stash, write the attachment for that
+        stash_data = {
+            'key_format': MLBF.KEY_FORMAT,
+            'stash_time': generation_time,
+            'attachment_type': BLOCKLIST_RECORD_STASH,
+        }
+        with storage.open(mlbf.stash_path, 'r') as stash_file:
+            attachment = ('stash.json', stash_file)
+            server.publish_attachment(stash_data, attachment)
```

Comment:
Can you publish the JSON directly instead of a file?

Using multiple attachments makes the client implementation more complicated. The [client library](https://searchfox.org/mozilla-central/rev/97cb0a90bd053de87cd1ab7646d5565809166bb1/services/settings/Attachments.jsm#62-74) takes the filename and overwrites existing files with the same name, so I would have to implement my own storage mechanism to avoid this issue.

Reply:
We could - I had them as attachments to cover the possible use-case (e.g. ESR) of turning off stash support so it wouldn't be downloading unnecessary data in the records.  If we're pretty confident that won't happen (and/or don't care about the extra download data if it does) then it can be directly in the data block.

Reply:
https://github.com/mozilla/addons-server/pull/13986/commits/a79c82f67ed89a36986fc2acd872e95db48ffc34